src/zenml/cli/service.py

```diff
@@ -0,0 +1,148 @@
+from zenml.services import ServiceRegistry, ServiceState
+from zenml.zen_service.zen_service import ZenService, ZenServiceConfig
+
+logger = get_logger(__name__)
+SERVICE_CONFIG_FILENAME = "ZenService.json"
+GLOBAL_ZENML_SERVICE_CONFIG_FILEPATH = os.path.join(
+    get_global_config_directory(), SERVICE_CONFIG_FILENAME
+)
+
+
+@cli.group()
+def service() -> None:
+    """ZenMl server."""
+
+
+@service.command("up", help="Start a daemon service running the zenml service")
+@click.option("--port", type=int, default=8000, show_default=True)
+@click.option("--profile", type=str, default=None)
+def up_server(port: int, profile: Optional[str]) -> None:
+    """Provisions resources for the zen service."""
+    if profile is not None:
+        profile_configuration = GlobalConfiguration().get_profile(profile)
+        if profile_configuration is None:
+            raise ValueError(f"Could not find profile of name {profile}")
+        service_config = ZenServiceConfig(
+            port=port, store_profile_configuration=profile_configuration
+        )
+    else:
+        service_config = ZenServiceConfig(port=port)
+
+    print(f"{service_config=}")
+    try:
+        with open(GLOBAL_ZENML_SERVICE_CONFIG_FILEPATH, "r") as f:
+            zen_service = ServiceRegistry().load_service_from_json(f.read())
+    except (JSONDecodeError, FileNotFoundError, ModuleNotFoundError):
+        zen_service = ZenService(service_config)
+
+    cli_utils.declare(
+        f"Provisioning resources for service "
+        f"'{zen_service.SERVICE_TYPE.name}'."
+    )
+
+    zen_service.start(timeout=30)
+
+    if zen_service.endpoint:
+        if zen_service.endpoint.status.port != port:
+            cli_utils.error(
+                f"You specified port={port} but the service is "
+                f"running at '{zen_service.endpoint.status.uri}' "
+                f"This can happen in case the specified port is "
+                f"in use or if the service was already running "
+                f"on port{zen_service.endpoint.status.port}' "
+                f"In case you want to change to port={port} "
+                f"shut down the service with "
+                f"`zenml service down -f` and restart it with "
+                f"a free port of your choice."
+            )
+        else:
+            cli_utils.declare(
+                "Zenml Service running at "
+                f"'{zen_service.endpoint.status.uri}'."
+            )
+    else:
+        raise ValueError("No endpoint found for Zen Service.")
+
+    with open(GLOBAL_ZENML_SERVICE_CONFIG_FILEPATH, "w") as f:
+        f.write(zen_service.json(indent=4))
+
+
+@service.command("status")
+def status_server() -> None:
+    """Provisions resources for the stack."""
+    try:
+        with open(GLOBAL_ZENML_SERVICE_CONFIG_FILEPATH, "r") as f:
+            zervice = ServiceRegistry().load_service_from_json(f.read())
+    except FileNotFoundError:
+        cli_utils.error("No service found!")
```

Comment:
Is this worth raising an error or can we get away by simply displaying a warning to the user? (Same in the `down` function)